lib/octopus_serverspec_extensions/type/octopus_deploy_account.rb

```diff
@@ -0,0 +1,126 @@
+require 'serverspec'
+require 'serverspec/type/base'
+require 'net/http'
+require 'json'
+
+module Serverspec::Type
+  class OctopusDeployAccount < Base
+    @account = nil
+    @serverUrl = nil
+    @apiKey = nil
+    @serverSupportsSpaces = nil
+    @spaceId = nil
+    @spaceFragment = ""
+
+    def initialize(serverUrl, apiKey, account_name, spaceID = 'Spaces-1')
+      @name = "Octopus Deploy Account #{account_name}"
+      @runner = Specinfra::Runner
+      @serverUrl = serverUrl
+      @apiKey = apiKey
+      @spaceId = spaceID
+
+      if serverUrl.nil?
+        raise "'serverUrl' was not provided. Unable to connect to Octopus server to validate configuration."
+      end
+      if apiKey.nil?
+        raise "'apiKey' was not provided. Unable to connect to Octopus server to validate configuration."
+      end
+      if account_name.nil?
+        raise "'account_name' was not provided. Unable to connect to Octopus server to validate configuration."
+      end
+
+      @serverSupportsSpaces = check_supports_spaces(serverUrl)
+
+      if @serverSupportsSpaces
+        @spaceFragment = "#{@spaceId}/"
+      end
+
+      @account = get_account_via_api(serverUrl, apiKey, account_name)
+    end
+
+    def exists?
+      (!@account.nil?) && (@account != [])
+    end
+
+    def has_description?(account_description)
+      return false if @account.nil?
+      @account["Description"] == account_description
+    end
+
+    def is_account_type?(account_type_name)
+      accounttypes = ['SshKeyPair', 'UsernamePassword', 'AzureSubscription', 'Token', 'AmazonWebServicesAccount']
+      if !accounttypes.include? account_type_name
+        raise("'#{account_type_name}' is not a valid account type")
+      end
+      return false if @account.nil?
+
+      @account["AccountType"] == account_type_name
+    end
+
+    def is_azure_account?
+      return false if @account.nil?
+      @account["AccountType"] == "AzureSubscription"
+    end
+
+    def is_aws_account?
+      return false if @account.nil?
+      @account["AccountType"] == "AmazonWebServicesAccount"
```

Comment:
maybe use a constant here? and in the methods above?

Reply:
I was kinda thinking it's a better fit for some kind of enum magic, but my Ruby is not exactly ninja level

like 

```
describe OctopusAccount($url, $apikey, 'my_user_account') do
   it { should be_account_type(OctopusAccount::AwsAccount) }
end

Or something. Actually, a public CONST would achieve much the same effect, wouldn't it?
```

Reply:
Constants have been added. I don't really know if they're ideal. ALso, they probably need to be documented, so I'll update the readme in this PR